pkg/config/v3/config_test.go

```diff
@@ -405,23 +405,23 @@ var _ = Describe("Cfg", func() {
 		})
 
 		It("DecodePluginConfig should fail for no plugin config object", func() {
-			var pluginCfg PluginConfig
-			err := c0.DecodePluginConfig(key, &pluginCfg)
+			var decodedPluginCfg PluginConfig
+			err := c0.DecodePluginConfig(key, &decodedPluginCfg)
 			Expect(err).To(HaveOccurred())
 			Expect(errors.As(err, &config.PluginKeyNotFoundError{})).To(BeTrue())
```

Comment:
Please Consider using the gomega's `MatchError` matcher.

Reply:
@nunnatsa You're absolutely right - this should be addressed. I'd prefer to keep it out of scope for this PR though, as it's unrelated to the main change. There are a few places in the codebase where `errors.As` is used in the tests, so it would make sense to handle all of them together in a separate PR. WDYT?